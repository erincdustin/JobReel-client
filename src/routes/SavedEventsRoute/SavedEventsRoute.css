.SavedEventsRoute {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 1em;
}

.savedEventsBack {
  position: relative;
  grid-column-start: 1;
  grid-column-end: 9;
}

.savedEventsBack > img {
  height: 250px;
  width: 250px;
  position: absolute;
  top: -200px;
  right: 5%;
  z-index: -1;
}

@media (max-width: 960px) {
  .SideNav{
    display: none;
  }
}

@media (min-width: 960px) {
  .TopNav{
    display: none;
  }
}