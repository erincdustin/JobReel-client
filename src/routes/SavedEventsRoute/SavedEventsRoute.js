import React from 'react';
import SideNav from '../../components/SideNav/SideNav';
import AddEventForm from '../../components/AddEventForm/AddEventForm';
import jobReelApiService from '../../services/jobreel-api-service';
import JobReelContext from '../../context/JobReelContext';

class SavedEventsRoute extends React.Component {
  
  static contextType = JobReelContext

  componentDidMount = () => {
    jobReelApiService.getSavedEvents()
      .then(res => {
        this.context.setSavedEvents(res);
      })
  }

  render() {
    console.log('Saved events:');
    console.log(this.context.savedEvents);
    return (
      <div className="SavedEventsRoute">
        <SideNav />
        <AddEventForm />
      </div>
    )
  }
}

export default SavedEventsRoute;