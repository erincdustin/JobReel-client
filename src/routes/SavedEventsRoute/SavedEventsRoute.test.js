import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import SavedEventsRoute from './SavedEventsRoute'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('SavedEventsRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><SavedEventsRoute /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('SavedEventsRoute snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(
      <SavedEventsRoute />
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})