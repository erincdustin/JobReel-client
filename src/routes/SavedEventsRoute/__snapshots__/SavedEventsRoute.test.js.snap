// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedEventsRoute snapshot test renders the UI as expected 1`] = `
<div
  className="SavedEventsRoute"
>
  <div
    className="title"
  >
    <h2>
      Events
    </h2>
  </div>
  <div
    className="savedEventsBack"
  >
    <img
      alt="saved-events-background"
      src="savedEventsBack.svg"
    />
  </div>
  <MediaQueryContextConsumer
    minDeviceWidth={961}
  >
    <SideNav />
  </MediaQueryContextConsumer>
  <MediaQueryContextConsumer
    maxDeviceWidth={960}
  >
    <TopNav />
  </MediaQueryContextConsumer>
  <AddEventForm />
</div>
`;
